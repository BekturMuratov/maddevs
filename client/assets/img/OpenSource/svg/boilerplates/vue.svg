<svg xmlns="http://www.w3.org/2000/svg" width="336" height="125" fill="none" xmlns:v="https://vecta.io/nano"><g clip-path="url(#A)"><path d="M206.784.001l-38.797 67.203L129.191.001H-.008l167.996 290.997L335.984.001h-129.2z" fill="#41b883"/><path d="M206.785.001l-38.797 67.203L129.191.001h-62l100.798 174.595L268.786.001h-62.001z" fill="#34495e"/></g><defs><clipPath id="A"><path fill="#fff" d="M0 0h336v291H0z"/></clipPath></defs></svg>