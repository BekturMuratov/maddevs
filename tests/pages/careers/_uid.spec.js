import { render } from '@testing-library/vue'
import Vacancy from '@/pages/careers/_uid.vue'
import { createLocalVue } from '@vue/test-utils'
import Vuelidate from 'vuelidate'

const store = {
  getters: {
    vacancy: () => ({
      labels: {
        remote: true,
        relocation: false,
      },
    }),
  },
}

const mocks = {
  $getMediaFromS3: () => 'img.jpg',
  $v: {
    name: {
      $touch: jest.fn(),
    },
    position: {
      $touch: jest.fn(),
    },
    grade: {
      $touch: jest.fn(),
    },
    email: {
      $touch: jest.fn(),
    },
    cvFile: {
      $touch: jest.fn(),
    },
    $reset: jest.fn(),
    validationGroup: {},
  },
  $store: {
    dispatch: jest.fn(),
  },
  buildEmail: jest.fn(),
  resetForm: jest.fn(),
  $refs: {
    fileInput: {
      reset: jest.fn(),
    },
    radioButtons: {
      reset: jest.fn(),
    },
  },
  $lazyLoad: {
    init: jest.fn(),
  },
  featureFlag: jest.fn(),
}

const localVue = createLocalVue()

localVue.use(Vuelidate)

describe('Careers _uid component', () => {
  it('should render correctly', () => {
    const { container } = render(Vacancy, {
      mocks,
      store,
    })

    expect(container).toMatchSnapshot()
  })
})
