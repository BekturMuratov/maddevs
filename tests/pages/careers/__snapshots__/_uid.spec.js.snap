// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Careers _uid component should render correctly 1`] = `
<div>
  <section
    class="careers-position"
    id="careers-position"
  >
    <div
      class="careers-position__header"
    >
      <div
        class="container"
      >
        <div
          class="careers-position__container"
        >
          <div
            class="careers-position__labels"
          >
            <div
              class="position-labels"
            >
              <div
                class="position-labels__item position-labels__item--filled"
              >
                
    Remote
  
              </div>
               
              <!---->
            </div>
          </div>
           
          <h1
            class="careers-position__title"
          >
            
        
      
          </h1>
           
          <!---->
           
          <!---->
           
          <button
            class="ui-button careers-position__apply-button"
            style="position: relative; overflow: hidden;"
            type="button"
          >
            
        Apply now
      
            <div
              class="wave"
            />
          </button>
        </div>
      </div>
    </div>
     
    <div
      class="container"
    >
      <div
        class="careers-position__container"
      >
        <div
          class="careers-position__info"
        >
          <!---->
           
          <div
            class="careers-position__benefits"
          >
            <h2
              class="careers-position__benefits-title"
            >
              
            Employees benefits
          
            </h2>
             
            <div
              class="careers-position__benefits-grid"
            >
              <div
                class="benefit-card"
              >
                <div
                  class="benefit-card__icon-wrapper"
                >
                  <img
                    alt="Hours Icon."
                    class="benefit-card__icon"
                    data-src="img.jpg"
                    height="30"
                    width="31"
                  />
                </div>
                 
                <p
                  class="benefit-card__title"
                >
                  Flexible working 
                  <br />
                   hours
                </p>
              </div>
              <div
                class="benefit-card"
              >
                <div
                  class="benefit-card__icon-wrapper"
                >
                  <img
                    alt="Trips Icon."
                    class="benefit-card__icon"
                    data-src="img.jpg"
                    height="32"
                    width="28"
                  />
                </div>
                 
                <p
                  class="benefit-card__title"
                >
                  Onsite business 
                  <br />
                   trips
                </p>
              </div>
              <div
                class="benefit-card"
              >
                <div
                  class="benefit-card__icon-wrapper"
                >
                  <img
                    alt="Lessons Icon."
                    class="benefit-card__icon"
                    data-src="img.jpg"
                    height="31"
                    width="36"
                  />
                </div>
                 
                <p
                  class="benefit-card__title"
                >
                  English 
                  <br />
                   lessons
                </p>
              </div>
              <div
                class="benefit-card"
              >
                <div
                  class="benefit-card__icon-wrapper"
                >
                  <img
                    alt="Training Icon."
                    class="benefit-card__icon"
                    data-src="img.jpg"
                    height="24"
                    width="39"
                  />
                </div>
                 
                <p
                  class="benefit-card__title"
                >
                  Training 
                  <br />
                   budget
                </p>
              </div>
              <div
                class="benefit-card"
              >
                <div
                  class="benefit-card__icon-wrapper"
                >
                  <img
                    alt="Culture Icon."
                    class="benefit-card__icon"
                    data-src="img.jpg"
                    height="31"
                    width="24"
                  />
                </div>
                 
                <p
                  class="benefit-card__title"
                >
                  Remote-first 
                  <br />
                   culture
                </p>
              </div>
              <div
                class="benefit-card"
              >
                <div
                  class="benefit-card__icon-wrapper"
                >
                  <img
                    alt="Docs Icon."
                    class="benefit-card__icon"
                    data-src="img.jpg"
                    height="30"
                    width="26"
                  />
                </div>
                 
                <p
                  class="benefit-card__title"
                >
                  Long-term 
                  <br />
                   projects
                </p>
              </div>
              <div
                class="benefit-card"
              >
                <div
                  class="benefit-card__icon-wrapper"
                >
                  <img
                    alt="Microphone Icon."
                    class="benefit-card__icon"
                    data-src="img.jpg"
                    height="33"
                    width="20"
                  />
                </div>
                 
                <p
                  class="benefit-card__title"
                >
                  Paid 
                  <br />
                   conferences
                </p>
              </div>
              <div
                class="benefit-card"
              >
                <div
                  class="benefit-card__icon-wrapper"
                >
                  <img
                    alt="Cocktail Glass Icon."
                    class="benefit-card__icon"
                    data-src="img.jpg"
                    height="36"
                    width="31"
                  />
                </div>
                 
                <p
                  class="benefit-card__title"
                >
                  Team 
                  <br />
                   parties
                </p>
              </div>
            </div>
          </div>
        </div>
         
        <div
          class="careers-position__contacts"
          id="careers-position-form"
        >
          <div
            class="hr-contact"
          >
            <div
              class="hr-contact__image"
            >
              <img
                alt="Diana"
                data-src="img.jpg"
              />
            </div>
             
            <div
              class="hr-contact__info"
            >
              <p
                class="hr-contact__title"
              >
                
      Hi, I'm Diana.
    
              </p>
               
              <p
                class="hr-contact__description"
              >
                
      Applying for jobs should be easy.
      I am here to simplify the hiring process and set you up for success.
      Feel free to send me a message on Telegram.
      
                <a
                  href="https://t.me/DianaHRDigital"
                  target="_blank"
                >
                  <img
                    data-src="img.jpg"
                    height="17"
                    width="20"
                  />
                   
                  <span>
                    @DianaHRDigital
                  </span>
                </a>
              </p>
            </div>
          </div>
           
          <div
            class="careers-position__form"
          >
            <div>
              <form
                class="position-form"
              >
                <div
                  class="position-form__message"
                >
                  <label
                    class="position-form__field"
                  >
                    
        Hello, my name is
        
                    <div
                      class="career-form-input position-form__input position-form__input-name"
                      title="Your Name"
                    >
                      <pre>
                        Your Name
                      </pre>
                       
                      <input
                        placeholder="Your Name"
                        type="text"
                      />
                    </div>
                    .
        
                    <br />
                     I want to work for you as a desired position.
         
                    <!---->
                  </label>
                   
                  <p
                    class="position-form__field"
                  >
                    <div
                      class="ui-radio-buttons position-form__positions-list"
                    >
                      <div
                        class="ui-radio-buttons_item"
                      >
                        <div
                          class="ui-radio-buttons_item-circle"
                        />
                         
                        <div
                          class="ui-radio-buttons_item-label"
                        >
                          
      Senior,
    
                        </div>
                      </div>
                      <div
                        class="ui-radio-buttons_item"
                      >
                        <div
                          class="ui-radio-buttons_item-circle"
                        />
                         
                        <div
                          class="ui-radio-buttons_item-label"
                        >
                          
      Middle,
    
                        </div>
                      </div>
                      <div
                        class="ui-radio-buttons_item"
                      >
                        <div
                          class="ui-radio-buttons_item-circle"
                        />
                         
                        <div
                          class="ui-radio-buttons_item-label"
                        >
                          
      Junior,
    
                        </div>
                      </div>
                      <div
                        class="ui-radio-buttons_item"
                      >
                        <div
                          class="ui-radio-buttons_item-circle"
                        />
                         
                        <div
                          class="ui-radio-buttons_item-label"
                        >
                          
      Intern
    
                        </div>
                      </div>
                    </div>
                     
                    <span>
                      <!---->
                    </span>
                  </p>
                   
                  <label
                    class="position-form__field"
                  >
                    
        Please reply to
        
                    <div
                      class="career-form-input position-form__input position-form__input-email"
                      title="your@mail.com"
                    >
                      <pre>
                        your@mail.com
                      </pre>
                       
                      <input
                        placeholder="your@mail.com"
                        type="email"
                      />
                    </div>
                     
                    <!---->
                  </label>
                   
                  <p
                    class="position-form__field position-form__skills"
                  >
                    
        To get more information on my skills, please
        
                    <ul
                      class="position-form__skills-list"
                    >
                      <li>
                        <label>
                          
              check out my
              
                          <div
                            class="career-form-input position-form__input position-form__input-linkedin"
                            title="LinkedIn profile"
                          >
                            <pre>
                              LinkedIn profile
                            </pre>
                             
                            <input
                              placeholder="LinkedIn profile"
                              type="text"
                            />
                          </div>
                        </label>
                        
            AND
          
                      </li>
                       
                      <li>
                        <label>
                          <div
                            class="file-input"
                          >
                            <input
                              accept=".pdf,.cv,.doc,.odt,.docx,.txt"
                              class="file-input__uploader"
                              data-testid="test-file"
                              id="file"
                              name="file"
                              type="file"
                            />
                             
                            <label
                              class="file-input__cv file-input__cv--attachable"
                              data-testid="test-file-label"
                              for="file"
                            >
                              
    find attached my CV
                              <span
                                class="red"
                              >
                                *
                              </span>
                            </label>
                          </div>
                           
                          <!---->
                        </label>
                      </li>
                    </ul>
                  </p>
                </div>
                 
                <button
                  class="ui-button"
                  style="position: relative; overflow: hidden;"
                  type="submit"
                >
                  
      I want to work for Mad Devs!
    
                  <div
                    class="wave"
                  />
                </button>
              </form>
               
              <transition-stub
                name="fade"
              >
                <!---->
              </transition-stub>
            </div>
          </div>
        </div>
      </div>
    </div>
  </section>
</div>
`;
